### DISCUSSION TITLE
How to wait for transition to finish before loading new page?

### DISCUSSION THREAD

**Discussion by goldentoaste** [sveltejs/kit#9013] [Q&A] [answered]
See here for the problem: [Sandbox](https://codesandbox.io/p/sandbox/elegant-shadow-zq2sgz?file=%2Fsrc%2Froutes%2F%2Bpage.svelte&selection=%5B%7B%22endColumn%22%3A1%2C%22endLineNumber%22%3A9%2C%22startColumn%22%3A1%2C%22startLineNumber%22%3A9%7D%5D)

I have 2 pages, each with an element that has a transition `fade`. When navigating to a different page, the fade animation will start playing and the content the new pages loads right away, before the fade is done yet. Is there anyway I can wait for the transition to finish?

**Comment by NormandoHall** [comment]
Removing `|local` ?

**Reply by goldentoaste** [reply]
From what I have seen, adding or removing `|local` does not seem to make a difference, even though this post suggests that having `|local` would fix the issue: [issue #6686](https://github.com/sveltejs/svelte/issues/6686). 

I think the semantics with `|local` is that the transition only plays when the component itself is destroyed, and not play if its destroyed because parent is destroyed, ie, when navigating to a new page. So I think if working as intended, `|local` would prevent the transition animation from playing, but the new page still loads instantly, which is not ideal, but better than the current situation.

**Comment by goldentoaste** [comment, marked as answer]
I think I have a fix for this issue, but if there is a better or easier way to do this I would love to know.  
[Fix here](https://codesandbox.io/p/sandbox/svelte-transition-blocking-demo-forked-m0m6tb?file=%2Fsrc%2Froutes%2F%2Blayout.svelte)

So the idea is to have a store to keep state if the current page needs to play outro, in a shared store. When a page change is triggered, cancel the nav, then trigger the nav again when outro is done playing. See comments in the code sandbox for details.

**Reply by ioucyf** [reply]
Does this work for you?

[codesandbox.io link](https://codesandbox.io/p/sandbox/svelte-transition-block-fix-forked-8lvk0g?file=%2Fsrc%2Flib%2Fcomponents%2FPageTransition.svelte&selection=%5B%7B%22endColumn%22%3A3%2C%22endLineNumber%22%3A17%2C%22startColumn%22%3A3%2C%22startLineNumber%22%3A17%7D%5D)

**Reply by goldentoaste** [reply]
@ofeenee Yup, this works great, thanks!
